phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
having another method for the reason is a bad design as you will have to implement the logic twice.
Another solution might to to deprecate is_enableable() and add another method which can return a string or an array of string. Then we can deal with both form to maintain BC (as we would have done if we have changed the return type of the existing method)